apis/v1beta1/collector_webhook.go

```diff
@@ -77,3 +90,17 @@ func (c CollectorWebhook) validate(r *OpenTelemetryCollector) (admission.Warning
 	}
 	return warnings, nil
 }
+
+func SetupCollectorWebhook(mgr ctrl.Manager, cfg config.Config, reviewer *rbac.Reviewer) error {
```

Comment:
Shouldn't we also add defaulter and validator logic for this webhook?

Reply:
There is already defaulting and validating webhook

Reply:
Can we add a simple E2E test using a v1beta1 Collector, to make sure this works as expected?